### PR TITLE
Upgrade `finch-tensor`

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi,

Once https://github.com/willow-ahrens/finch-tensor/pull/45 is merged and released this PR can go in. 
I also moved `pytest-xdist` to the test dependencies, WDYT?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -163,7 +163,7 @@ jobs:
         python-version: '3.11'
     - name: Install build and test dependencies from PyPI
       run: |
-        python -m pip install pytest-xdist -r array-api-tests/requirements.txt
+        python -m pip install -r array-api-tests/requirements.txt
     - name: Build and install Sparse
       run: |
         python -m pip install '.[finch]'
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:169]: > I also moved pytest-xdist to the test dependencies, WDYT?

That won't work, unless we do this too.

```suggestion
        python -m pip install '.[finch,tests]'
```

**Review by hameerabbasi — COMMENTED** [review]
Let's also add `-n $(nproc)` or whatever to each CI job.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mtsokol) [reply .github/workflows/ci.yml:169]: Ah, right. On the second thought let's drop these changes as they're not relevant to `finch-tensor` update and leave it for "light CI refactor" PR.

Comment (hameerabbasi) [reply .github/workflows/ci.yml:169]: That sounds like a good idea to keep changes isolated.

**Review by hameerabbasi — APPROVED** [review]

**Review by hameerabbasi — DISMISSED** [review]

**Comment by mtsokol** [comment]
Maybe we can cut a new alpha release after merging it?

**Comment by hameerabbasi** [comment]
> Maybe we can cut a new alpha release after merging it?

Done: https://pypi.org/project/sparse/0.16.0a8/

**Pull request merged**